python/ray/node.py

```diff
@@ -400,6 +402,16 @@ def _get_unused_port(self):
         s.bind(("", 0))
         port = s.getsockname()[1]
         s.close()
+
+        # Try to generate a port that is far above the 'next available' one
+        for _ in range(NUMBER_OF_PORT_RETRIES):
```

Comment:
I don't really want to have an infinite loop here, so I instead decided to try a few times and then return the original port. I don't imagine this loop will execute all 10 times, but wanted to err on the side of caution. 
Should I: 
1) Leave as a constant
2) Make a configurable option
3) Make this a `while True` loop

Reply:
@ijrsvt can we also make sure that gRPC doesn't segfault in these cases?

Reply:
The current approach seems fine to me